File: BeeBreeder.Common/AlleleDatabase/Bee/Adaptation.cs
namespace BeeBreeder.Common.AlleleDatabase.Bee
{
    public readonly struct Adaptation
    {
        public readonly int Up;
        public readonly int Down;

        public Adaptation(int up = 0, int down = 0)
        {
            Up = up;
            Down = down;
        }

        public override string ToString()
        {
            if (Up == Down)
            {
                if (Up == 0 )
                {
                    return "0";
                }
                return $"+-{Up}";
            }

            var up = Up == 0 ? "" : $"+{Up}";
                var down = Down == 0 ? "" : $"-{Down}";
                return $"{up}{down}";
            }
        }
    }

File: BeeBreederWeb/src/app/model/bee/adaptation.ts
export class Adaptation {
  public up : number
  public down : number


  constructor(up: number = 0, down: number = 0) {
    this.up = up;
    this.down = down;
  }

  public toString() : string {
    if (this.up == this.up)
    {
      if (this.up == 0 )
      {
        return "0";
      }
      return '+-${this.up}';
    }

    let up = this.up == 0 ? "" : "+${this.up}";
    let down = this.down == 0 ? "" : "-${this.down}";
    return "${up}${down}";
  }
}
